File: scriptingcsharp/csharp/Scripting/Backup/Host/Scripting.cs
using System;
using System.CodeDom.Compiler;

namespace Host
{
	public class Scripting
	{

		public enum Languages
		{
			VB,
			CSharp
		}

		public static CompilerResults CompileScript(string Source, string Reference, Languages Language)
		{
			CodeDomProvider provider = null;

			switch(Language)
			{
				case Languages.VB:
					provider = new Microsoft.VisualBasic.VBCodeProvider();
					break;
				case Languages.CSharp:
					provider = new Microsoft.CSharp.CSharpCodeProvider();
					break;
			}

			return CompileScript(Source, Reference, provider);
		}

		public static CompilerResults CompileScript(string Source, string Reference, CodeDomProvider Provider)
		{
			ICodeCompiler compiler = Provider.CreateCompiler();
			CompilerParameters parms = new CompilerParameters();
			CompilerResults results;

			// Configure parameters
			parms.GenerateExecutable = false;
			parms.GenerateInMemory = true;
			parms.IncludeDebugInformation = false;
			if (Reference != null && Reference.Length != 0)
				parms.ReferencedAssemblies.Add(Reference);
			parms.ReferencedAssemblies.Add("System.Windows.Forms.dll");
			parms.ReferencedAssemblies.Add("System.dll");

			// Compile
			results = compiler.CompileAssemblyFromSource(parms, Source);

			return results;
		}

		public static object FindInterface(System.Reflection.Assembly DLL, string InterfaceName)
		{
			// Loop through types looking for one that implements the given interface
			foreach(Type t in DLL.GetTypes())
			{
				if (t.GetInterface(InterfaceName, true) != null)
					return DLL.CreateInstance(t.FullName);
			}

			return null;
		}

	}
}

File: scriptingcsharp/vb/Scripting/Host/Scripting.vb
Imports System.CodeDom.Compiler

Public Class Scripting

    Public Enum Languages
        VB
        CSharp
    End Enum

    Public Shared Function CompileScript(ByVal Source As String, ByVal Reference As String, ByVal Language As Languages) As CompilerResults
        Dim provider As CodeDomProvider = Nothing

        Select Case Language
            Case Languages.VB
                provider = New Microsoft.VisualBasic.VBCodeProvider()
            Case Languages.CSharp
                provider = New Microsoft.CSharp.CSharpCodeProvider()
        End Select

        Return CompileScript(Source, Reference, provider)
    End Function

    Public Shared Function CompileScript(ByVal Source As String, ByVal Reference As String, ByVal Provider As CodeDomProvider) As CompilerResults
        Dim compiler As ICodeCompiler = Provider.CreateCompiler()
        Dim params As New CompilerParameters()
        Dim results As CompilerResults

        'Configure parameters
        With params
            .GenerateExecutable = False
            .GenerateInMemory = True
            .IncludeDebugInformation = False
            If Not Reference Is Nothing AndAlso Reference.Length <> 0 Then .ReferencedAssemblies.Add(Reference)
            .ReferencedAssemblies.Add("System.Windows.Forms.dll")
            .ReferencedAssemblies.Add("System.dll")
        End With

        'Compile
        results = compiler.CompileAssemblyFromSource(params, Source)

        Return results
    End Function

    Public Shared Function FindInterface(ByVal DLL As Reflection.Assembly, ByVal InterfaceName As String) As Object
        Dim t As Type

        'Loop through types looking for one that implements the given interface
        For Each t In DLL.GetTypes()
            If Not (t.GetInterface(InterfaceName, True) Is Nothing) Then
                Return DLL.CreateInstance(t.FullName)
            End If
        Next

        Return Nothing
    End Function

End Class
